pages/docs/learn/versioning.mdx

```diff
@@ -0,0 +1,39 @@
+export const description = "Learn how ."
+
+# Versioning long running functions
+
+It's inevitable that long-running functions change over time. Inngest allows developers to implement multiple strategies for changing long-running code over time. In order to understand how to best manage changes over time, it's important to understand how the SDK implements determinism and [executes steps](/docs/learn/how-functions-are-executed).
+
+## Determinism in functions
+
+Determinism is the same in *every* Inngest language SDK. Except for language-specific idioms, all SDKs implement the same logic. In all SDKs, functions in Inngest are a series of steps. Each step runs reliably, will retry on failure, and is as close to exactly once as possible (outbound network failure when reporting completing steps notwithstanding).
+
+### How the SDK works with steps
+
+As covered in [_How Inngest functions are executed_](/docs/learn/how-functions-are-executed), each step in a function has a unique identifier, represented as a string. Each time a step is found, the SDK checks whether the step has been executed. It does this by:
+
+1. Hashing the step's unique identifier along with an counter of the number of times the step has been called. This allows steps to be used in a loop.
```

Comment:
```suggestion
1. Hashing the step's unique identifier along with a counter of the number of times the step has been called. In this way, steps can be used in a loop.
```

Reply:
```suggestion
1. Hashing the step's unique identifier along with a counter of the number of times the step has been called. This enables steps to be used in a loop.
```
Riff on the edit for clarity of what this does